packages/eslint-plugin/src/rules/consistent-type-imports.ts

```diff
@@ -183,12 +183,15 @@ export default createRule<Options, MessageIds>({
                        * keep origin import kind when export
                        * export { Type }
                        * export default Type;
+                       * export = Type;
                        */
                       if (
                         ref.identifier.parent.type ===
                           AST_NODE_TYPES.ExportSpecifier ||
                         ref.identifier.parent.type ===
-                          AST_NODE_TYPES.ExportDefaultDeclaration
+                          AST_NODE_TYPES.ExportDefaultDeclaration ||
+                        ref.identifier.parent.type ===
+                          AST_NODE_TYPES.TSExportAssignment
```

Comment:
Thanks for the review!

---

> I believe here should be no change required to the rule here - just a fix to the scope analysis. The tests are correct to add though!

If we don't check `TSExportAssignment` here, then cases such as

```ts
import type A from 'foo';
export = A;
```

will result in false positives. From my understanding, export assignment here behaves like default export (exports type/value as is).

Please correct me if I'm wrong

Reply:
ah you're right.
I'd just assumed that the rule would handle these dual-type-value references and not enforce anything for the import - I didn't remember that it was special-cased.